src/libstd/collections/hash/map.rs

```diff
@@ -353,37 +352,50 @@ fn search_hashed<K, V, M, F>(table: M,
     // undefined behavior when Bucket::new gets the raw bucket in this
     // case, immediately return the appropriate search result.
     if table.capacity() == 0 {
-        return TableRef(table);
+        return InternalEntry::TableIsEmpty;
```

Comment:
Ah, sorry - I just didn't see the second group when I wrote that comment. Given the presence of accessors, returning `TableIsEmpty` makes sense.